packages/block-editor/src/hooks/auto-inserting-blocks.js

```diff
@@ -0,0 +1,235 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { addFilter } from '@wordpress/hooks';
+import { Fragment } from '@wordpress/element';
+import { PanelBody, ToggleControl } from '@wordpress/components';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { createBlock, store as blocksStore } from '@wordpress/blocks';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { InspectorControls } from '../components';
+import { store as blockEditorStore } from '../store';
+
+function AutoInsertingBlocksControl( props ) {
+	const { autoInsertedBlocksForCurrentBlock, groupedAutoInsertedBlocks } =
+		useSelect(
+			( select ) => {
+				const { getBlockTypes } = select( blocksStore );
+				const _autoInsertedBlocksForCurrentBlock =
+					getBlockTypes()?.filter(
+						( block ) =>
+							block.autoInsert &&
+							Object.keys( block.autoInsert ).includes(
+								props.blockName
+							)
+					);
+
+				// Group by block namespace (i.e. prefix before the slash).
+				const _groupedAutoInsertedBlocks =
+					_autoInsertedBlocksForCurrentBlock?.reduce(
+						( groups, block ) => {
+							const [ prefix ] = block.name.split( '/' );
```

Comment:
is `"namespace"` reserved or in use at this point? we even mention it in the comment, but then call it `prefix` - is there a reason not to leave the name in place as `namespace` or `blockNamespace`?

Reply:
Good point 👍 And no underlying reason for not using it 😅

Reply:
0204ddad0c